samples/sensor/tmp108/src/main.c

```diff
@@ -141,6 +142,10 @@ int main(void)
 		printf("warning: tmp108 device not found checking for compatible ams device\n");
 
 		temp_sensor = DEVICE_DT_GET_ANY(ams_as6212);
+	}
+
+	if (!temp_sensor) {
+		temp_sensor = DEVICE_DT_GET_ANY(ams_as6221);
 
```

Comment:
It is the same thing of course, but this would make more sense to me as

```suggestion
		printf("warning: tmp108 device not found checking for compatible ams device\n");

		temp_sensor = DEVICE_DT_GET_ANY(ams_as6212);
		if (!temp_sensor)
			temp_sensor = DEVICE_DT_GET_ANY(ams_as6221);

```

Reply:
I'd rather avoid the nested ifs, especially since more devices can be added here in the future.